src/config/section/storage.js

```diff
@@ -325,6 +326,16 @@ export default {
       details: ['name', 'id', 'displayname', 'description', 'type', 'current', 'parentName', 'virtualmachineid', 'account', 'domain', 'created'],
       searchFilters: ['name', 'domainid', 'account', 'tags'],
       actions: [
+        {
+          api: 'createSnapshotFromVMSnapshot',
+          icon: 'camera',
+          label: 'label.action.create.snapshot.from.vmsnapshot',
+          message: 'message.action.create.snapshot.from.vmsnapshot',
+          dataView: true,
+          popup: true,
+          show: (record) => { return record.state === 'Ready' },
```

Comment:
@ravening Could you add a check whether the instance hypervisor is KVM since the legacy UI only supports it for KVM.
https://github.com/apache/cloudstack/blob/master/ui/scripts/storage.js#L3220

Reply:
@davidjumani I dont find the key `hypervisor` in `record`. So it always returns undefined when I try to access `record.hypervisor`. Any idea on this?

Reply:
@ravening The hypervisor can be found via the `virtualmachineid` in the record

Reply:
@ravening @davidjumani can we make it simpler then? maybe change the list response to pass hypervisor and use it here. Note: 1.0/GA ships with 4.15 - so if it makes it simpler, send a PR against 4.14 branch and return the key as part of vmsnapshot list response?

Reply:
@rhtyd if we have to return hypervisor in list snapshot response then we need to call listvirtualmachine api for each snpashot response which might increase response time and also burden on mgt server


As @davidjumani suggested, we have virtulamachineid parameter but we need to call listvirtualmachine api to get the hypervisor type. Im trying to make that work

Reply:
@ravening That might not work since api calls are async and will return after the `show` method has completed execution. It would be better to alter the API response in the backend to include the vm's hypervisor which could be done quite easily without calling further apis.

Reply:
@davidjumani @rhtyd so then I will just check `record.hypervisor === 'KVM'` here and add `hypervisor` field to listVMSnapshot response in backend.

I still need to call list virtual machine to get the hypervisor type

Reply:
@ravening Yes, add that condition, once the backend has been changed to return the hypervisor as well, we can merge this!
There won't be a need to call another API as the hypervisor field will be added in the backend itself

Reply:
@davidjumani I have updated the code here. In the meantime I will work on adding hypervisor type in the backend code

Reply:
@ravening `record.state === 'Ready' && record.hypervisor === 'KVM'` :smile: